### DISCUSSION TITLE
Text in a parallelepiped

### DISCUSSION THREAD

**Discussion by SizNi** [cduck/drawSvg#112] [Q&A] [answered]
Hello! I wanted to ask for your help: I have different sized parallelepipeds inside which I want to write text of different lengths. Is there a similar functionality?

**Comment by cduck** [comment]
Hi. I'm not exactly sure what you are trying to do. In general, if you provide a short code sample of what you've tried and a sketch of what you are trying to draw it will be easier for others to help.

A parallelepiped is a 3D shape but SVGs only represent 2D shapes.  You will need to [manually calculate the projected 2D coordinates](https://en.wikipedia.org/wiki/3D_projection) of the eight corners (or find a library that can do this).  Then you can use `draw.Lines(...)` to draw the projected view of each face.

If you want the text to be centered, you'll need to calculate the 2D coordinate of the text center and use the centering options of `draw.Text(...)`.  Any length text is allowed.

**Reply by SizNi** [reply]
Yes indeed, my mistake, I meant rectangle. The problem is that, for example, I have a text long 50 characters, and there is a rectangle, which in one line fits only 20 characters, respectively, if I want to write text in the rectangle, I need a transfer on the lines inside it, for example, as in a cell table MS Office or as a figure in Autocad. Maybe there is such functionality?

**Reply by cduck** [reply, marked as answer]
I don't know what you mean by "transfer on the lines inside it".  `drawsvg.Text` supports multiple lines of text (see examples) but SVGs in general do not support automatic word wrapping.

**Reply by SizNi** [reply]
Ok, i understand. Tank you for your lib!
